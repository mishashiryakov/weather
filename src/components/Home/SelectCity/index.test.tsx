import React from "react";
import { shallow, render, mount } from "enzyme";
import { SelectCity } from "./index";

const setSelectedCityMock = jest.fn();

const MOCK_PROPS = {
  cities: ["Minsk", "Moscow", "Bratislava"],
  setSelectedCity: setSelectedCityMock,
  selectedCity: "Minsk",
};

describe("SelectCity component", () => {
  it("should render 3 buttons", function () {
    const wrapper = shallow(<SelectCity {...MOCK_PROPS} />);
    expect(wrapper.find("button")).toHaveLength(3);
  });

  it("should match snapshot with 3 buttons", function () {
    const wrapper = shallow(<SelectCity {...MOCK_PROPS} />);
    expect(wrapper).toMatchSnapshot();
  });
  
  it("should call setSelectedCity function on button Click", () => {
    const wrapper = shallow(<SelectCity {...MOCK_PROPS} />);
    wrapper.find('.selected').simulate('click');
    expect(setSelectedCityMock).toHaveBeenCalledTimes(1)
  })
});
