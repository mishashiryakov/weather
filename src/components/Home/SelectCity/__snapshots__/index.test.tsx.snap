// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectCity component should match snapshot with 3 buttons 1`] = `
<div
  className="selectCityButtons"
>
  <button
    className="selected"
    key="Minsk"
    onClick={[Function]}
  >
    Minsk
  </button>
  <button
    className=""
    key="Moscow"
    onClick={[Function]}
  >
    Moscow
  </button>
  <button
    className=""
    key="Bratislava"
    onClick={[Function]}
  >
    Bratislava
  </button>
</div>
`;
